File: test_complete_auth_flow.js
// Complete authentication flow test
const testCompleteAuthFlow = async () => {
  console.log('🔍 Testing complete authentication flow...');
  
  const baseUrl = 'http://localhost:8000';
  
  try {
    // Step 1: Login and get tokens
    console.log('\n1. 🔐 Testing login...');
    const loginResponse = await fetch(`${baseUrl}/api/auth/login`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: 'testuser_1758806039@example.com',
        password: 'testpassword123'
      }),
    });
    
    if (!loginResponse.ok) {
      const errorData = await loginResponse.json();
      console.error('❌ Login failed:', errorData);
      return;
    }
    
    const loginData = await loginResponse.json();
    const accessToken = loginData.tokens.access_token;
    console.log('✅ Login successful for:', loginData.user.email);
    
    // Step 2: Test /auth/me endpoint (used by BackendAuthContext)
    console.log('\n2. 👤 Testing /auth/me endpoint...');
    const meResponse = await fetch(`${baseUrl}/api/auth/me`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (meResponse.ok) {
      const userData = await meResponse.json();
      console.log('✅ /auth/me successful:', userData.email);
    } else {
      const errorData = await meResponse.json();
      console.error('❌ /auth/me failed:', errorData);
      return;
    }
    
    // Step 3: Test Dashboard endpoints
    console.log('\n3. 📊 Testing Dashboard endpoints...');
    
    // Test getUserAnalytics (the failing one)
    console.log('   Testing /analytics/user...');
    const userAnalyticsResponse = await fetch(`${baseUrl}/api/analytics/user`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (userAnalyticsResponse.ok) {
      const analyticsData = await userAnalyticsResponse.json();
      console.log('✅ /analytics/user successful');
    } else {
      const errorData = await userAnalyticsResponse.json();
      console.log('❌ /analytics/user failed:', errorData.message);
    }
    
    // Test getJobAnalytics
    console.log('   Testing /analytics/jobs...');
    const jobAnalyticsResponse = await fetch(`${baseUrl}/api/analytics/jobs`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (jobAnalyticsResponse.ok) {
      const jobData = await jobAnalyticsResponse.json();
      console.log('✅ /analytics/jobs successful');
    } else {
      const errorData = await jobAnalyticsResponse.json();
      console.log('❌ /analytics/jobs failed:', errorData.message);
    }
    
    // Step 4: Test Upload page endpoints
    console.log('\n4. 📤 Testing Upload page endpoints...');
    
    // Test user settings (used by Upload page)
    const settingsResponse = await fetch(`${baseUrl}/api/user/settings`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (settingsResponse.ok) {
      console.log('✅ /user/settings successful');
    } else {
      const errorData = await settingsResponse.json();
      console.log('❌ /user/settings failed:', errorData.message);
    }
    
    // Step 5: Test History page endpoints
    console.log('\n5. 📜 Testing History page endpoints...');
    
    // Test user history
    const historyResponse = await fetch(`${baseUrl}/api/history?page=1&limit=10`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (historyResponse.ok) {
      const historyData = await historyResponse.json();
      console.log('✅ /history successful, jobs:', historyData.jobs?.length || 0);
    } else {
      const errorData = await historyResponse.json();
      console.log('❌ /history failed:', errorData.message);
    }
    
    // Test jobs endpoint
    const jobsResponse = await fetch(`${baseUrl}/api/jobs`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (jobsResponse.ok) {
      const jobsData = await jobsResponse.json();
      console.log('✅ /jobs successful, count:', jobsData.length || 'N/A');
    } else {
      const errorData = await jobsResponse.json();
      console.log('❌ /jobs failed:', errorData.message);
    }
    
    // Step 6: Test Settings page endpoints
    console.log('\n6. ⚙️ Testing Settings page endpoints...');
    
    // Test user profile
    const profileResponse = await fetch(`${baseUrl}/api/user/profile`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (profileResponse.ok) {
      console.log('✅ /user/profile successful');
    } else {
      const errorData = await profileResponse.json();
      console.log('❌ /user/profile failed:', errorData.message);
    }
    
    // Test user stats
    const statsResponse = await fetch(`${baseUrl}/api/users/stats`, {
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json',
      },
    });
    
    if (statsResponse.ok) {
      console.log('✅ /users/stats successful');
    } else {
      const errorData = await statsResponse.json();
      console.log('❌ /users/stats failed:', errorData.message);
    }
    
    console.log('\n🎉 Authentication flow test completed!');
    console.log('\n📋 Summary:');
    console.log('- Authentication system: ✅ Working');
    console.log('- Token validation: ✅ Working');
    console.log('- Some API endpoints have validation issues (not auth-related)');
    console.log('\n💡 The "No active session" errors are likely due to:');
    console.log('1. Frontend not properly storing/retrieving tokens from localStorage');
    console.log('2. BackendAuthContext not properly initializing on app startup');
    console.log('3. API service not properly sending auth headers');
    
  } catch (error) {
    console.error('💥 Test failed with error:', error);
  }
};

// Run the test
testCompleteAuthFlow();

File: test_complete_auth_flow.py
#!/usr/bin/env python3
"""
Comprehensive test script to verify the complete authentication flow
including frontend login and Results page access
"""

import requests
import json
import time

def test_backend_frontend_health():
    """Test that both backend and frontend are running"""
    print("🔍 Testing Backend and Frontend Health")
    print("=" * 50)
    
    # Test backend
    try:
        backend_response = requests.get("http://localhost:8000/health", timeout=5)
        if backend_response.status_code == 200:
            print("✅ Backend is running")
        else:
            print(f"❌ Backend health check failed: {backend_response.status_code}")
            return False
    except requests.exceptions.RequestException as e:
        print(f"❌ Backend is not accessible: {e}")
        return False
    
    # Test frontend
    try:
        frontend_response = requests.get("http://localhost:3000", timeout=5)
        if frontend_response.status_code == 200:
            print("✅ Frontend is running")
            return True
        else:
            print(f"❌ Frontend health check failed: {frontend_response.status_code}")
            return False
    except requests.exceptions.RequestException as e:
        print(f"❌ Frontend is not accessible: {e}")
        return False

def test_api_authentication():
    """Test authentication via API"""
    print("\n🔐 Testing API Authentication")
    print("=" * 50)
    
    test_user = {
        "email": "testuser1758805676@example.com",
        "password": "TestPassword123!"
    }
    
    try:
        # Test login
        login_response = requests.post(
            "http://localhost:8000/api/auth/login",
            json=test_user,
            headers={"Content-Type": "application/json"},
            timeout=10
        )
        
        if login_response.status_code == 200:
            result = login_response.json()
            access_token = result.get('tokens', {}).get('access_token')
            
            if access_token:
                print("✅ API login successful")
                print(f"   User: {result.get('user', {}).get('email', 'N/A')}")
                
                # Test protected endpoint
                headers = {"Authorization": f"Bearer {access_token}"}
                me_response = requests.get(
                    "http://localhost:8000/api/auth/me",
                    headers=headers,
                    timeout=10
                )
                
                if me_response.status_code == 200:
                    print("✅ Protected endpoint access successful")
                    return True, access_token, test_user
                else:
                    print(f"❌ Protected endpoint failed: {me_response.status_code}")
                    return False, None, None
            else:
                print("❌ No access token received")
                return False, None, None
        else:
            print(f"❌ API login failed: {login_response.status_code}")
            print(f"   Response: {login_response.text}")
            return False, None, None
            
    except requests.exceptions.RequestException as e:
        print(f"❌ API authentication test failed: {e}")
        return False, None, None

def test_frontend_authentication_simple():
    """Test frontend authentication without browser automation"""
    print("\n🌐 Testing Frontend Authentication (Simple)")
    print("=" * 50)
    
    # Test that key frontend routes are accessible
    routes = [
        ("/", "Home page"),
        ("/login", "Login page"),
        ("/register", "Register page"),
        ("/results", "Results page")
    ]
    
    all_accessible = True
    
    for route, description in routes:
        try:
            response = requests.get(f"http://localhost:3000{route}", timeout=5)
            if response.status_code == 200:
                print(f"✅ {description} accessible")
            else:
                print(f"⚠️  {description} returned: {response.status_code}")
                if route == "/results":
                    # Results page might redirect if not authenticated, which is expected
                    print("   (This might be expected if authentication is required)")
        except requests.exceptions.RequestException as e:
            print(f"❌ {description} failed: {e}")
            all_accessible = False
    
    return all_accessible

def test_results_page_content():
    """Test that the Results page content is accessible"""
    print("\n📊 Testing Results Page Content")
    print("=" * 50)
    
    try:
        # Try to access the results page directly
        response = requests.get("http://localhost:3000/results", timeout=10)
        
        if response.status_code == 200:
            content = response.text
            
            # Check if the page contains expected content or if it's a redirect
            if "<!DOCTYPE html>" in content and "results" in content.lower():
                print("✅ Results page content accessible")
                
                # Check for common error indicators
                if "white screen" in content.lower() or "error" in content.lower():
                    print("⚠️  Potential issues detected in page content")
                    return False
                else:
                    print("✅ No obvious errors in page content")
                    return True
            else:
                print("⚠️  Results page might be redirecting or showing unexpected content")
                return False
        else:
            print(f"⚠️  Results page returned status: {response.status_code}")
            # This might be expected if authentication is required
            return True
            
    except requests.exceptions.RequestException as e:
        print(f"❌ Results page test failed: {e}")
        return False

def run_comprehensive_test():
    """Run all tests in sequence"""
    print("🚀 Starting Comprehensive Authentication Flow Test")
    print("=" * 60)
    
    # Test 1: Health checks
    if not test_backend_frontend_health():
        print("\n❌ Health checks failed. Cannot proceed.")
        return False
    
    # Test 2: API authentication
    api_success, access_token, test_user = test_api_authentication()
    if not api_success:
        print("\n❌ API authentication failed. Cannot proceed.")
        return False
    
    # Test 3: Frontend routes
    frontend_success = test_frontend_authentication_simple()
    if not frontend_success:
        print("\n⚠️  Some frontend routes had issues, but continuing...")
    
    # Test 4: Results page content
    results_success = test_results_page_content()
    
    # Summary
    print("\n" + "=" * 60)
    print("📋 COMPREHENSIVE TEST SUMMARY")
    print("=" * 60)
    
    if api_success and frontend_success and results_success:
        print("🎉 ALL TESTS PASSED!")
        print("\n✅ Backend authentication: WORKING")
        print("✅ Frontend routes: ACCESSIBLE")
        print("✅ Results page: ACCESSIBLE")
        
        print("\n🔑 Test Credentials for Manual Verification:")
        print(f"   Email: {test_user['email']}")
        print(f"   Password: {test_user['password']}")
        
        print("\n💡 Manual Testing Steps:")
        print("   1. Open http://localhost:3000/login")
        print("   2. Login with the above credentials")
        print("   3. Navigate to http://localhost:3000/results")
        print("   4. Verify no white screen appears")
        
        return True
    else:
        print("❌ SOME TESTS FAILED")
        print(f"   API Authentication: {'✅' if api_success else '❌'}")
        print(f"   Frontend Routes: {'✅' if frontend_success else '❌'}")
        print(f"   Results Page: {'✅' if results_success else '❌'}")
        
        return False

if __name__ == "__main__":
    success = run_comprehensive_test()
    
    if success:
        print("\n🎯 Authentication system appears to be working correctly!")
        print("   The white screen issue should be resolved.")
    else:
        print("\n🔧 Some issues detected. Please check the output above.")